include/rmm/rmm.hpp

```diff
@@ -205,10 +202,12 @@ inline rmmError_t realloc(T** ptr, size_t new_size, cudaStream_t stream,
 inline rmmError_t free(void* ptr, cudaStream_t stream, const char* file,
                    unsigned int line) {
   rmm::LogIt log(rmm::Logger::Free, ptr, 0, stream, file, line);
-  if (rmm::Manager::usePoolAllocator())
-    RMM_CHECK_CNMEM(cnmemFree(ptr, stream));
-  else
-    RMM_CHECK_CUDA(cudaFree(ptr));
+  if(ptr != nullptr){
```

Comment:
There shouldn't be a need to check for `nullptr`. The resources should be able to handle calling `deallocate` on nullptr. This is tested here: https://github.com/rapidsai/rmm/blob/branch-0.10/tests/mr/mr_tests.cpp#L126